scipy/interpolate/_rbfinterp.py

```diff
@@ -0,0 +1,538 @@
+
+    if kernel not in _AVAILABLE:
+        raise ValueError('`kernel` must be one of %s' % _AVAILABLE)
+
+    if epsilon is None:
+        if kernel in _SCALE_INVARIANT:
+            epsilon = 1.0
+        else:
+            raise ValueError(
+                '`epsilon` must be specified if `kernel` is not one of %s' %
+                _SCALE_INVARIANT
+                )
+    else:
+        epsilon = float(epsilon)
+
+    min_degree = _NAME_TO_MIN_DEGREE.get(kernel, -1)
+    if degree is None:
+        degree = max(min_degree, 0)
+    else:
+        degree = int(degree)
+        if degree < -1:
+            raise ValueError('`degree` must be at least -1')
+        elif degree < min_degree:
+            warnings.warn(
+                '`degree` should not be below %d when `kernel` is "%s". The '
+                'interpolant may not be uniquely solvable, and the smoothing '
+                'parameter may have an unintuitive effect.' %
+                (min_degree, kernel),
+                UserWarning
+                )
+
+    if k is None:
+        nobs = ny
+    else:
+        # make sure the number of nearest neighbors used for interpolation does
+        # not exceed the number of observations
+        k = int(min(k, ny))
+        nobs = k
+
+    # The polynomial matrix must have full column rank in order for the
+    # interpolant to be well-posed, which is not possible if there are fewer
+    # observations than monomials
+    nmonos = int(binom(degree + ndim, ndim))
+    if nmonos > nobs:
+        raise ValueError(
+            'At least %d data points are required when `degree` is %d and the '
+            'number of dimensions is %d. Consider lowering `degree` and/or '
+            'setting `kernel` to an RBF with a lower minimum degree.' %
+            (nmonos, degree, ndim)
+            )
+
+    return y, d, smoothing, kernel, epsilon, degree, k
+
+
+class RBFInterpolator:
+    """
+    Radial basis function (RBF) interpolation in N dimensions
+
+    Parameters
+    ----------
+    y : (P, N) array_like
+        Data point coordinates
+    d : (P, ...) array_like
+        Data values at `y`
+    smoothing : float or (P,) array_like, optional
+        Smoothing parameter. The interpolant perfectly fits the data when this
+        is set to 0. For large values, the interpolant approaches a least
+        squares fit of a polynomial with the specified degree.
+    kernel : str, optional
+        Type of RBF. This should be one of:
+
+            - 'linear'                       : ``-r``
+            - 'tps' (thin plate spline)      : ``r**2 * log(r)``
+            - 'cubic'                        : ``r**3``
+            - 'quintic'                      : ``-r**5``
+            - 'mq' (multiquadratic)          : ``-sqrt(1 + r**2)``
+            - 'imq' (inverse multiquadratic) : ``1/sqrt(1 + r**2)``
+            - 'iq' (inverse quadratic)       : ``1/(1 + r**2)``
+            - 'ga' (Gaussian)                : ``exp(-r**2)``
```

Comment:
Could we not spell out everything? `gaussian` instead of `ga`, etc. There is also the possibility to have something similar as in `spatial.distance.cdist` where there are aliases. I would not necessarily do this here, but I prefer to be explicit. Also who knows if in x years some of these short names become slang words. Cf our issue with `trapz, simps`.

Reply:
My concern about using the full names is that there is a greater likelihood of misspelling them. For example, would it be "thin plate spline", "thin_plate_spline", "thin-plate_spline", etc.? I also just realized that I have been misspelling "multiquadric" as "multiquadratic".  I imagine that would be a common mistake if the user needed to write out the full name.  

AFAIK the abbreviations that I am using here are commonly used in the RBF literature.  So if they do eventually become slang terms, we can point to the literature to justify their usage. I image that is why we are still using the acronym "RBF"...

I think my preference is to use aliases here, where the user can specify the full names (with underscores for spaces) or the abbreviations.

Reply:
I also would like to prevent abbreviation far-west like here https://github.com/scipy/scipy/pull/13955/files#r622687974 (here the problem is that `lambda` cannot be used. Still, someone could want to abbreviate differently).

`simps` is also widely used but still we had to change it #12924.

As for which orthography, IMO there is just one possibility as we are in Python, so `thin_plate_spline`.

I would not block anything here, just a recommendation to be explicit here 😄

Reply:
I replaced the abbreviations with the full names, and I am not using aliases (although the names are case-insensitive as suggested above)